### PR TITLE
Upgrade the forked Home IdP extension to latest commit

### LINKED ISSUE #83: Update home idp lookup
to current version https://github.com/sventorben/keycloak-home-idp-discovery

### PR DESCRIPTION
**Pull request by phamann**
### TL;DR
Upgrades the forked version of https://github.com/sventorben/keycloak-home-idp-discovery embded within the projects source to the latest commit and then re-applies the nesscary keycloak-orgs patches ontop. 

The main driving factor is to pickup improvements in the extension since we forked and take advantge of new configuration parameters.

### Notes for reviewers
To help with the review I've isolated the custom p2-inc/keycloak-org changes to the extension in the second commit ~~https://github.com/p2-inc/keycloak-orgs/commit/323cb0607113d4e2209caf69c642d65a2a60a4d2~~ 4cb622f9a16487c8718f521c1774ef304ef23a5f

---

closes #83

### CODE REVIEW

**Comment by phamann** [comment]
I've just realised I didn't check to see if there was a diff in any of the template resources or messages. Will try push an update today.

**Comment by phamann** [comment]
Have now, pushed template changes 👍🏻

**Review by xgp — APPROVED** [review]

**Comment by xgp** [comment]
@phamann Everything looks good. I'd eventually like to figure out a way to test this, but I don't yet have a good way of testing frontend flows like this. I ran some manual tests on the branch locally, and I see no regressions.

File: src/main/java/io/phasetwo/service/auth/idp/HomeIdpDiscoveryAuthenticatorFactory.java [hunk 1]
```diff
@@ -16,81 +11,86 @@
...
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
 
-  private static final Logger LOG = Logger.getLogger(HomeIdpDiscoveryAuthenticatorFactory.class);
+import static org.keycloak.models.AuthenticationExecutionModel.Requirement.*;
 
-  private static final AuthenticationExecutionModel.Requirement[] REQUIREMENT_CHOICES =
-      new AuthenticationExecutionModel.Requirement[] {REQUIRED, ALTERNATIVE, DISABLED};
+public final class HomeIdpDiscoveryAuthenticatorFactory implements AuthenticatorFactory, ServerInfoAwareProviderFactory {
```
Comment (xgp) [inline src/main/java/io/phasetwo/service/auth/idp/HomeIdpDiscoveryAuthenticatorFactory.java:20]: @phamann One thing I had to add to get it working was to add back in the `AutoService` annotation. I use this to generate the `META-INF/services` file. Can you add back? Here is the diff I made locally:
```
diff --git a/src/main/java/io/phasetwo/service/auth/idp/HomeIdpDiscoveryAuthenticatorFactory.java b/src/main/java/io/phasetwo/service/auth/idp/HomeIdpDiscoveryAuthenticatorFactory.java
index 0d6258b..263c976 100755
--- a/src/main/java/io/phasetwo/service/auth/idp/HomeIdpDiscoveryAuthenticatorFactory.java
+++ b/src/main/java/io/phasetwo/service/auth/idp/HomeIdpDiscoveryAuthenticatorFactory.java
@@ -1,6 +1,7 @@
 //package de.sventorben.keycloak.authentication.hidpd;
 package io.phasetwo.service.auth.idp;
 
+import com.google.auto.service.AutoService;
 import org.jboss.logging.Logger;
 import org.keycloak.Config;
 import org.keycloak.authentication.Authenticator;
@@ -17,6 +18,7 @@ import java.util.Map;
 
 import static org.keycloak.models.AuthenticationExecutionModel.Requirement.*;
 
+@AutoService(AuthenticatorFactory.class)
 public final class HomeIdpDiscoveryAuthenticatorFactory implements AuthenticatorFactory, ServerInfoAwareProviderFactory {
 
     private static final Logger LOG = Logger.getLogger(HomeIdpDiscoveryAuthenticatorFactory.class);
```

**Review by xgp — CHANGES_REQUESTED** [review]
Add back the autoservice annotation, please. Otherwise LGTM!

File: src/main/java/io/phasetwo/service/auth/idp/HomeIdpDiscoveryAuthenticatorFactory.java (hunk 1 above)
Comment (phamann) [reply src/main/java/io/phasetwo/service/auth/idp/HomeIdpDiscoveryAuthenticatorFactory.java:22]: Woops, good catch. Have fixed in: https://github.com/p2-inc/keycloak-orgs/pull/121/commits/b925e461e61e9e5113701ae04e66a240e777a884

**Comment by phamann** [comment]
@xgp sorry for the delay here. I've now fixed the `AutoService` annotation.
